worker.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"log"
+	"time"
+
+	"fmt"
+
+	"github.com/zalando-incubator/kube-ingress-aws-controller/aws"
+	"github.com/zalando-incubator/kube-ingress-aws-controller/kubernetes"
+)
+
+func startPolling(awsAdapter *aws.Adapter, kubernetesClient *kubernetes.Client, pollingInterval time.Duration) {
+	for {
+		il, err := kubernetes.ListIngress(kubernetesClient)
+		if err != nil {
+			log.Println(err)
+		} else {
+			arns := flattenIngressByARN(il)
+			if missingARNs := filterExistingARNs(awsAdapter, arns); len(missingARNs) > 0 {
+				for missingARN, ingresses := range missingARNs {
+					lb, err := createMissingLoadBalancer(awsAdapter, missingARN)
+					if err != nil {
+						log.Println(err)
+						break
+					}
+					// TODO: attach LoadBalancer to AutoScalingGroup ?
+					log.Printf("successfully created ALB %q for certificate %q\n", lb.ARN(), missingARN)
+
+					if err := kubernetes.UpdateIngressLoaBalancer(kubernetesClient, ingresses, lb); err != nil {
+						log.Println(err)
+					} else {
+						log.Printf("updated ingresses %v with DNS name %q\n", ingresses, lb.DNSName())
+					}
+				}
+			}
+		}
+
+		time.Sleep(pollingInterval)
+	}
+}
+
+func flattenIngressByARN(il *kubernetes.IngressList) map[string][]kubernetes.Ingress {
+	uniqueARNs := make(map[string][]kubernetes.Ingress)
```

Comment:
we should consider the cases when no arn is set.

as a general comment: to avoid crashing with panic, it is a good practice to be full defensive with parsed objects that come from an API, even if the API specification guarantees that, e.g. .Metadata, will be always there in a successful response.